### PR TITLE
[Work In Progress - Do Not Merge] Connectedk8s: Add "arc "mode in Guard for using OBO service

### PR DESCRIPTION
**Pull request by vineeth-thumma**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/connectedk8s/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 Release History
 
+1.3.21
+++++++
+
+* Setting default authnMode to 'arc' when RBAC is enabled
```
Comment (julienstroheker) [inline src/connectedk8s/HISTORY.rst:8]: `Setting guard default authnMode to `arc`....`

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 2]
```diff
@@ -1469,8 +1465,7 @@ def enable_features(cmd, client, resource_group_name, cluster_name, features, ku
         cmd_helm_upgrade.extend(["--kube-context", kube_context])
     if enable_azure_rbac:
         cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.enabled=true"])
-        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientId={}".format(azrbac_client_id)])
-        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientSecret={}".format(azrbac_client_secret)])
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.authnMode=arc"])  # new mode for latest OBO changes
```
Comment (julienstroheker) [inline src/connectedk8s/azext_connectedk8s/custom.py:1468]: you can remove the `# new mode for latest OBO changes` comment and add a better one on top of this flag to explain this change

**Review by julienstroheker — CHANGES_REQUESTED** [review]
Blocking PR for now. Since we want to wait before this getting merged.

File: src/connectedk8s/HISTORY.rst (hunk 1 above)
Comment (vineeth-thumma) [reply src/connectedk8s/HISTORY.rst]: Updated here: https://github.com/Azure/azure-cli-extensions/pull/6675

File: src/connectedk8s/azext_connectedk8s/custom.py (hunk 2 above)
Comment (vineeth-thumma) [reply src/connectedk8s/azext_connectedk8s/custom.py]: Updated here: https://github.com/Azure/azure-cli-extensions/pull/6675

**Comment by vineeth-thumma** [comment]
Closed this in favor of https://github.com/Azure/azure-cli-extensions/pull/6675

**Pull request closed without merging**